suitcase/tiff/__init__.py

```diff
@@ -101,7 +112,15 @@ def export(gen, filepath, **kwargs):
                     stream_name = stream_names[event_page['descriptor']]
                     for field in event_page['data']:
                         for img in event_page['data'][field]:
-                            files[event_page['descriptor']].save(img, *kwargs)
+                            # check that the data is 2D, if not raise exception
+                            if len(numpy.asarray(img).shape) == 2:
+                                files[event_page['descriptor']].save(img,
+                                                                     *kwargs)
+                            else:
+                                raise NonSupportedDatatype('one or more of the'
```

Comment:
Quibble on the name `NonSupportedDatatype`: the issue is with the _shape_ (dimensionality) not the _type_ (interpretation of bytes).

Can the error message include `field` and `ndim`? Using f-strings instead of `format` will make it easier to do this succinctly, if you like.

You can avoid making a long skinny column in the situation by doing it like:

```py
raise NonSupportedDatatype(
    "One or more of the...")
```

The guideline is, "If you include any arguments on the first line, immediately after the paren, everything has to be indented all the way to be even with the paren. But if you put no arguments on the first line, just indent everything four spaces."

Reply:
changed to NonSupportedDataShape()